[IMP] ecommerce: doc about orders paid with Payment Acquirers (#915)
odoo/documentation: ecommerce/shopper_experience/payment_acquirer.rst

================================================
How To Manage Orders Paid With Payment Acquirers
================================================

Odoo automatically confirms orders as soon as the payment is authorized 
by a payment acquirer. This triggers the delivery.
If you invoice based on ordered quantities,
you are also requested to invoice the order.

What Are The Different Payment Statuses
=======================================

> In Debug mode, the user can check the transaction status from the order, by selecting

Reviewer: This could use a link to `general/developer_mode/activate.rst`

> *Payment Transactions* under the *Configuration* menu.

Reviewer: This is not true. Users need to be in debug mode to see this menu item.

> .. image:: media/payment-transactions.png

Reviewer: What happened to the previous image file? If it's no longer used it should be removed.

Reviewer: And yet, in this particular case, it shouldn't be removed. Instead, it should be renamed to `payment-transactions.png`.
This is because git doesn't behave like a simple system folder. Instead of just holding the most recent record of the repository content, it remembers *everything*. So, renaming a file actually adds load (equal to the weight of the image file) on the repository. If we rename image files to often, we'll end-up with a multi-GB repository that will take ages to download.

And, if the user clicks on the link to the journal entry, they will be taken to that journal
entry's page, showcasing all the pertinent details surrounding this specific transaction

.. image:: media/reconciled-journal-entry.png
   :align: center
   :alt: payment is confirmed in the chatter of sales order

.. note:: Specific messages are provided to your customers for every
   payment status whenever they are redirected to Odoo after the transaction.
   To edit these messages, go to the *Messages* tab of the payment
   method.

Automatically generate invoices at order
========================================

When the order is confirmed, you can also have an invoice automatically issued
and paid. This fully-automated feature is designed for businesses that invoice 
orders straight on.

To do this, go to the *Invoicing* section of the *Settings* in the Website application,
and select *Invoice what is ordered* and then select *Automatic Invoices*, which will
generate an invoice automatically when the online payment is confirmed.

.. image:: media/automatic-invoice.png
   :align: center
   :alt: example of automatic invoice

.. note::

>    If you choose this mode, you are requested to select a payment journal, in order to record 
>    payments in your books. This payment is automatically reconciled with the invoice, marking it as
>    paid. Select your **bank account** if you get paid immediately on your bank account. If you don't 
>    you can create a specific journal for the payment acquirer (type = Bank). That way, you can track
>    online payments in an intermediary account of your books until you get paid into your bank
>    account (see :doc:`../../general/payment_acquirers/payment_acquirers`).

Reviewer 2: This note is not about about generating invoices automatically.
I believe it is linked to the next point "capture payment". Also, I think that the part about creating new journals for payment acquirers isn't completely correct anymore, since you would configure which journal to use when you first activate a new payment acquirer. (also, the new journal is now created automatically by odoo, when you do that).

I would simply remove this note.
```suggestion
```

> Capture payment after the delivery
> ==================================
> With this mode (which is only available with Authorize.net), the order is confirmed, 
> but the amount is kept on hold. Once the delivery is processed, you can capture the payment from
> Odoo.

Reviewer 2: - as at line 12: add a blank line between two commands (title and paragraph)
- break the line at 100th character
- "from Odoo" --> "from the related SO"
```suggestion
Capture payment after the delivery
==================================

With this mode (which is only available with Authorize.net), the order is confirmed, but the amount
is kept on hold. Once the delivery is processed, you can capture the payment from the related Sales
Order.
```

> To learn more about that, simply follow the procedure explained `here <https://www.odoo
> .com/documentation/user/14.0/general/payment_acquirers/authorize
> .html#capture-the-payment-after-the-delivery>`_.

Reviewer 2: Never use external links to target an internal doc or a specific part of a doc. Also, try to add the text on a meaningful description of the link rather than on "here" or "click here".

Instead, you can use either a `:doc:` directive or an internal hyperlink target. (See [RST cheat sheet: hyperlinks](https://www.odoo.com/documentation/user/14.0/contributing/documentation/rst_cheat_sheet.html#hyperlinks))

In this case, since the purpose of the paragraph is to refer to other docs, we can replace it with a [seealso](https://www.odoo.com/documentation/user/14.0/contributing/documentation/rst_cheat_sheet.html#seealso) section.

```suggestion
.. seealso::
   - :doc:`../../general/payment_acquirers/payment_acquirers`
   - :ref:`Payment Acquirers: Place a hold on a card <payment_acquirers/capture_amount>`
   - :doc:`../../general/payment_acquirers/authorize`
```

![image](https://user-images.githubusercontent.com/46810843/116718403-36a9e280-a9da-11eb-9be5-2f9729c60a8b.png)

=========================================
Manage orders paid with Payment Acquirers
=========================================

Odoo automatically confirms orders as soon as the payment is authorized 
by a payment acquirer. This triggers the delivery.
If you invoice based on ordered quantities, you are also requested to invoice the order.

Checking the status of a payment
================================
In `Debug <https://odoo.com/documentation/user/14.0/general/developer_mode/activate.html>`_
mode, the user can check the transaction status from the order, by selecting

> *Payment Transactions* under the *Configuration* menu.

Reviewer 2: Developer/debug mode is not the normal flow. If it's hidden from users, it's probably for a reason. There may be additional views hidden from most users but that doesn't mean it's the normal flow. In my understanding of SO, the payment is confirmed with an automatic note in the chatter, and if you create an invoice, the payment is directly reconciled.

I guess - not sure about it - that in older versions of odoo, the transaction was directly shown on the SO as the current screenshot suggests. These states were probably visible then.

Author: Ah, I see what you mean. Makes sense. So, seeing as how this is only accessible through Debug mode, and you mentioned that this first point was no longer relevant anyway, I will go ahead and remove this portion of the documentation. But, I will mention how the payment is confirmed in the Chatter of an SO (via an automatic note). And, I will also mention that if they create an invoice, a payment will be directly reconciled immediately.

.. image:: media/payment-transactions.png

>     :align: center

Reviewer 2: fix identation + add alt tags to all images

and select *Invoice What Is Ordered* and then select *Automatic Invoices*, which will
generate an invoice automatically when the online payment is confirmed.

.. image:: media/automatic-invoice.png
    :align: center

.. note::
   If you choose this mode, you are requested to select a payment journal, in order to record 
   payments in your books. This payment is automatically reconcilied with the invoice, marking it as 
   paid. Select your **bank account** if you get paid immediately on your bank account. If you don't 
   you can create a specific journal for the payment acquirer (type = Bank). That way, you can track
   online payments in an intermediary account of your books until you get paid into your bank
   account (see :doc:`../../general/payment_acquirers/payment_acquirers`).

Capture payment after the delivery
==================================
With this mode (which is only available with Authorize.net), the order is confirmed, 
but the amount is kept on hold. Once the delivery processed, you can capture the payment from Odoo.

.. image:: media/capture-manually.png
    :align: center

> To capture the payment, open the transaction from the order in *Payment Transactions*.

Reviewer 2: This is not in the hidden payment transactions menu anymore.
You can read more about it here https://www.odoo.com/documentation/user/14.0/general/payment_acquirers/authorize.html#capture-the-payment-after-the-delivery

Author: Great point! For the sake of efficiency, I'll simply add a quick link to this documentation, so as to avoid any unnecessary re-explaining. If this is the incorrect move, however, feel free to let me know how I should go about fixing this issue. Thanks again for all your continued patience and feedback during this all-too-important learning process. I sincerely appreciate it.

Then, click *Capture Transaction*.

> .. image:: media/payment-capture.png

Reviewer 2: This image looks quite small once the doc is built
